stablehlo/dialect/StablehloOps.cpp

```diff
@@ -4168,13 +4168,31 @@ LogicalResult TransposeOp::inferReturnTypes(
                              " of [",
                              range, "] but got ", permutation);
 
+  llvm::SmallVector<int64_t, 4> inputBounds(rank, ShapedType::kDynamicSize);
+  bool hasBounds = false;
+  if (auto encoding =
+          rankedTy.getEncoding().dyn_cast_or_null<TypeExtensionsAttr>()) {
+    inputBounds = llvm::to_vector<4>(encoding.getBounds());
+    hasBounds = true;
+  }
```

Comment:
It is possible that we put this part as an util function in the future. It looks good to check as-is now, as the code-sharing refactoring is on-going and this can be refined later if more ops will have some parts like this.

Reply:
sounds good.